projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-checkout.ts

```diff
@@ -475,21 +563,46 @@ export function interceptPaymentTypesEndpoint(): string {
   return alias;
 }
 
-export function interceptCheckoutB2BDetailsEndpoint(fixture: string) {
-  const alias = 'getCheckoutDetails';
-  cy.intercept(
-    {
-      method: 'GET',
-      path: `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
-        'BASE_SITE'
-      )}/users/**/carts/**/*?fields=deliveryAddress(FULL),deliveryMode(FULL),paymentInfo(FULL),costCenter(FULL),purchaseOrderNumber,paymentType(FULL)*`,
-    },
-    { fixture, statusCode: 200 }
-  ).as(alias);
+export function interceptCostCenterEndpoint() {
+  const alias = 'getCostCenters';
+
+  cy.intercept({
+    method: 'GET',
+    path: `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
+      'BASE_SITE'
+    )}/costcenters?fields=DEFAULT*`,
+  }).as(alias);
 
   return alias;
 }
 
+export function interceptCheckoutB2BDetailsEndpoint(
+  body?: any,
+  addressId?: string
+) {
+  const alias = 'getCheckoutDetails';
+  const request = {
+    method: 'GET',
+    path: `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
+      'BASE_SITE'
+    )}/users/**/carts/**/*?fields=deliveryAddress(FULL),deliveryMode(FULL),paymentInfo(FULL),costCenter(FULL),purchaseOrderNumber,paymentType(FULL)*`,
+  };
+
+  if (body && addressId) {
+    if (JSON.stringify(body).includes('addressIdFromServer')) {
+      body = JSON.parse(
```

Comment:
another solution to replace  'addressIdFromServer' value occurences is to go through all object values. I just thought 'stringify->replace->parse' was easier, won't be any side-effect as the JS object alreday defined from our sample-data.